subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,113 @@
+project(
+  'speexdsp',
+  'c',
+  version : '1.2.0',
+  license : 'bsd-like',
+  default_options : [
+    'c_std=c99',
+    'buildtype=release',
+    'default_library=both',
+    'b_lto=true',
+    'warning_level=0'])
+
+# library versioning from speexdsp-1.2.0's configure.ac
+# SPEEXDSP_LT_CURRENT=6, SPEEXDSP_LT_REVISION=1, SPEEXDSP_LT_AGE=5
+lib_current = 6
+lib_revision = 1
+lib_age = 5
+lib_soversion = lib_current - lib_age
+lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
+
+c_args = []
+cc = meson.get_compiler('c')
+host_cpu_family = host_machine.cpu_family()
+
+# Round up configuration data
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+cdata = configuration_data()
+
+# Set SIMD config data and flags
+simd_type = 'None'
+if host_cpu_family.startswith('x86')
+  cdata.set('USE_SSE', true)
+  cdata.set('USE_SSE2', true)
+  c_args += cc.get_supported_arguments(['-mfpmath=sse', '-msse2'])
+  simd_type = 'SSE2'
+
+elif host_cpu_family == 'arm'
+  cdata.set('USE_NEON', true)
+  c_args += cc.get_supported_arguments(['-mfpu=neon-vfpv4'])
+  simd_type = 'NEON (vfpv4)'
+
+elif host_cpu_family == 'aarch64'
+  cdata.set('USE_NEON', true)
+  c_args += cc.get_supported_arguments(['-mfpu=neon-fp-armv8'])
+  simd_type = 'NEON (armv8)'
+endif
+
+# Set numerical representation config data
+numerical_type = ''
+fft_library = ''
+if get_option('use-fixed-point')
+  # the fixed-point implementation doesn't use SSE intrinsics (and arch.h
+  # checks for this), however we still let the compiler pack floats into
+  # SIMD calls instructions whenever its cost model permits. 
+  cdata.set('USE_SSE', false)
+  cdata.set('FIXED_POINT', true)
+
+  numerical_type = 'fixed-point'
+  cdata.set('USE_KISS_FFT', true)
+  fft_library = 'Kiss FFT (built-in)'
+
+else
+  cdata.set('FLOATING_POINT', true)
+  numerical_type = 'floating-point'
+
+  cdata.set('USE_SMALLFT', true)
+  fft_library = 'OggVorbis FFT (built-in)'
+endif
+
+# Set C99 variable-length array config data
+cdata.set('VAR_ARRAYS', cc.get_id() != 'msvc')
+
+# Set EXPORT config data
+if cc.get_id() == 'msvc' and get_option('default_library') != 'static'
+  cdata.set('EXPORT', '')
+elif cc.has_function_attribute('visibility')
+  cdata.set('EXPORT', 'extern __attribute__ ((visibility ("default")))')
+else
+  cdata.set('EXPORT', 'extern')
+endif
+
+# Populate config.h from the above data
+configure_file(input : 'config.h.meson',
+               output : 'config.h',
+               configuration : cdata)
+
+# Add accumulated flags to the build
+foreach arg : c_args
+  add_project_arguments(arg, language: 'c')
+endforeach
```

Comment:
I don't think the foreach loop here is necessary, you can pass an array of options to `add_project_arguments`. Also, you could avoid to accumulate options in an array, just call `add_project_arguments` directly when discovering the required options.

Reply:
Thanks; now passing in the `c_args` array, and also performing a single `get_supported_arguments` call here too.  So that lightens the load up-top.

I could have added `add_project_arguments` and `get_supported_arguments` at each point in the code that involves a flag, however that's a bit more cumbersome on the eyes (and repetitive), so I prefer accumulating a simple list of values across the entire `.build` followed by a single point that checks and adds the arguments. 

(It fits with other wraps, and gives future contributors a bit less load to deal with if they're adding a couple more features).

Reply:
I'm perfectly okay with accumulating with an array, yeah.